packages/react-native-web/src/exports/Image/index.js

```diff
@@ -94,56 +100,105 @@ function getFlatStyle(style, blurRadius, filterId) {
-          ? curr
-          : prev
+    resolvedSource = resolveNumericSource(source);
+  } else if (typeof source === 'string') {
+    resolvedSource = resolveStringSource(source);
+  } else if (Array.isArray(source)) {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'The <Image> component does not support multiple sources passed as array, falling back to the first source in the list',
+        { source }
       );
     }
-    const scaleSuffix = scale !== 1 ? `@${scale}x` : '';
-    uri = asset
-      ? `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}`
-      : '';
-  } else if (typeof source === 'string') {
-    uri = source;
+
+    return resolveSource(source[0]);
   } else if (source && typeof source.uri === 'string') {
-    uri = source.uri;
+    resolvedSource = resolveObjectSource(source);
   }
 
-  if (uri) {
-    const match = uri.match(svgDataUriPattern);
-    // inline SVG markup may contain characters (e.g., #, ") that need to be escaped
+  if (resolvedSource.uri) {
+    const match = resolvedSource.uri.match(svgDataUriPattern);
     if (match) {
-      const [, prefix, svg] = match;
-      const encodedSvg = encodeURIComponent(svg);
-      return `${prefix}${encodedSvg}`;
+      resolvedSource = resolveSvgDataUriSource(resolvedSource, match);
     }
   }
 
-  return uri;
+  return resolvedSource;
+}
+
+// get the URI from the packager
+function resolveNumericSource(source: number): ImageSource {
+  const asset = getAssetByID(source);
+  if (asset == null) {
+    throw new Error(
+      `Image: asset with ID "${source}" could not be found. Please check the image source or packager.`
+    );
+  }
+  let scale = asset.scales[0];
+  if (asset.scales.length > 1) {
+    const preferredScale = PixelRatio.get();
+    // Get the scale which is closest to the preferred scale
+    scale = asset.scales.reduce((prev, curr) =>
+      Math.abs(curr - preferredScale) < Math.abs(prev - preferredScale)
+        ? curr
+        : prev
+    );
+  }
+
+  const scaleSuffix = scale !== 1 ? `@${scale}x` : '';
+  const uri = `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}`;
+  const { height, width } = asset;
+
+  return { uri, height, width };
+}
+
+function resolveStringSource(source: string): ImageSource {
+  return { uri: source };
+}
+
+function resolveObjectSource(source: Object): ImageSource {
+  return (source: ImageSource);
+}
+
+function resolveSvgDataUriSource(
+  source: Object,
+  match: Array<string>
+): ImageSource {
+  const [, prefix, svg] = match;
+  // inline SVG markup may contain characters (e.g., #, ") that need to be escaped
+  const encodedSvg = encodeURIComponent(svg);
+
+  return {
+    ...source,
+    uri: `${prefix}${encodedSvg}`
+  };
+}
+
+// resolve any URI that might have a local blob URL created with `createObjectURL`
+function resolveBlobUri(source: ImageSource): string {
+  return ImageLoader.resolveBlobUri(source.uri);
+}
```

Comment:
I'm having trouble piecing together why we need a function that just calls another function. But also unsure whether we are adding the ability to resolve a URI with a local blob in this PR. Is that one of the requirements? How is it related to adding "headers"?

Reply:
Examine how `ImageLoader` handles sources using headers and you will see that we do use `createObjectURL` with a `blob`, in such cases we need to resolve the blob created for the given `source.uri`